### ISSUE TITLE
Submit R bindings to CRAN

### ISSUE THREAD

**Issue by rcurtin** [mlpack/mlpack#2599]
Hey @yashwants19, @eddelbuettel, and @coatless,

It's really exciting that the R bindings are merged now!  @Yashwants19, you've done an amazing job this summer and we finally---after probably 6-8 years!---have R bindings automatically generated for mlpack.  I guess, the last step is submission to CRAN.  Releasing mlpack 3.4.0 is next on my list, so, that'll be ready in the next day or so and we can use that version for submission.

I know my name is as the maintainer, so is there anything I need to do here?  I can be instructed but as I am sure we are all familiar with now, I know very little about R other than how to copy-paste examples. :smile:

Let me know what you think.

(I figured, since the R bindings are now merged, the appropriate place for discussion is the main mlpack repository! :))

**Comment by eddelbuettel** [comment]
Awesomeness.  Consider this as a standing offer to aide you with the CRAN submission and everything, and feel free to hit me on other channels if ever there something not deemed "public".  

Basically the next steps are to (speed-)read the "Policy" you will agree to adhere to:  https://cran.r-project.org/web/packages/policies.html

And then to upload.  Let @coatless and my myself maybe have another go at the tarball to make sure it doesn't tickle any obvious WARNING or NOTE issues.  But @Yashwants19 has been so good and diligent that there should be few surprises.  And if and when you're ready you go to https://cran.r-project.org/submit.html, do an email handshake and then ... on a "new" package (RcppMLPACK does not count, different name, different maintainer) wait a couple of days.

**Comment by Yashwants19** [comment]
Thank you for opening the issue.

**Comment by rcurtin** [comment]
I got through the process of setting up R-devel and did `R CMD check --as-cran` on the tarball generated by the Github action (super easy!) but the release of Armadillo 9.900.3 breaks us and forces #2602 to be merged and thus 3.4.1 to be released before I can pass that check.  That should be just a day or two until all that is done. :+1:

**Comment by eddelbuettel** [comment]
How so? What breaks?  (And in the future once it is "up" it will of course by one 700+ reverse depends and checked prior to release.)  9.900.3 is a small delta over 9.900.2, and I had nothing else break so surprised it caused a wobble.  Or was it simply because you had the version number hardwired or something like that?

(Also, r-devel can be had via Docker containers I look after. Easier than building.  Same of course for R.)

**Comment by conradsnicta** [comment]
@eddelbuettel Long story short, due to this: https://github.com/mlpack/mlpack/issues/2596. Simple fix is in https://github.com/mlpack/mlpack/pull/2597, though the more elaborate fix in https://github.com/mlpack/mlpack/pull/2602 is preferred.

**Comment by eddelbuettel** [comment]
Thanks for the clarification. (And to be plain, in my world that is not "breakage" (as no functionality was "borked", or I missed something glancing at the tickets) but an extension / refactoring.  Hence my astonishment about the sudden "breakage".  Not so.)

**Comment by rcurtin** [comment]
@eddelbuettel it actually is a breakage---Armadillo 9.900.3 now marks `guess_file_type_internal()` as private, so mlpack currently won't compile until #2602 is merged (see #2601).

**Comment by eddelbuettel** [comment]
Missed that, thanks.

**Comment by rcurtin** [comment]
Submitted today; I'll update this issue when I hear back from CRAN. :+1:

**Comment by eddelbuettel** [comment]
Awesome.  One word of caution is that you .... need to be very patient.  One can look into incoming (I use a shell snippet via the old-yet-powerful `ncftp` client; there is also a [browseable site here](https://lockedata.github.io/cransays/articles/dashboard.html)) and see that some items take time to be processed. I uploaded RcppSpdlog on or around Aug 25, got one first piece of feedback ("more explicit author and copyright info, please") on Sep 1 and pushed back an update the same day and ... am still waiting.  Subsequent updates will be quicker.

**Comment by eddelbuettel** [comment]
BTW don't see it yet in incoming.  Make sure you also the email handshake.  From what I can tell `mlpack` is not yet in the queue.

**Comment by coatless** [comment]
"E-mail handshake" here is a single clickable URL that verifies you did indeed send a package to CRAN. The e-mail will always be sent to whoever is the maintainer of the _R_ package. 

Given `mlpack`'s complexities, I'm going to assume we'll probably hear back from them at the end of September.

**Comment by eddelbuettel** [comment]
Good reminder so I just checked.  Should be good:

https://github.com/mlpack/mlpack/blob/e11c3e320c4008a3c6ed16abeef69d53c0584366/src/mlpack/bindings/R/mlpack/DESCRIPTION.in#L6

**Comment by coatless** [comment]
Looks like it moved into the `pretest` queue. So, we're a quarter of the way there! 

![Screen Shot 2020-09-07 at 5 18 54 PM](https://user-images.githubusercontent.com/833642/92418673-51b7a180-f12e-11ea-9be0-ce8263e60fe3.png)

**Comment by rcurtin** [comment]
Awesome---I'll update here when I hear back from a CRAN maintainer. :+1:

**Comment by eddelbuettel** [comment]
Just checked again. You "survived" `pre-test` (a testament to the quality of the package) and are now at the very end of `newbies`.  That will take some time but you can always `ls -lt` in directory `newbies` to see how many are ahead of you.  It'll probably be a week.

**Comment by eddelbuettel** [comment]
I see you migrated to `archived/` so you must have gotten a love letter explaining why.  Feel free to share if you think we can help.

**Comment by rcurtin** [comment]
```
Date: Wed, 9 Sep 2020 09:04:57 +0200                                                                                                                                            
From: Uwe Ligges <ligges@statistik.tu-dortmund.de>                                                                                                                              
To: Ryan Curtin <ryan@ratml.org>, CRAN <cran-submissions@r-project.org>                                                                                                         
Subject: Re: CRAN submission mlpack 3.4.1                                                                                                                                       
X-Spam-Level:                                                                                                                                                                   
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:60.0) Gecko/20100101 Thunderbird/60.9.1                                                                                 
                                                                                                                                                                                
Thanks, we see:                                                                                                                                                                 
                                                                                                                                                                                
  License components with restrictions and base license permitting such:                                                                                                        
    BSD_3_clause + file LICENSE                                                                                                                                                 
  File 'LICENSE':                                                                                                                                                               
    mlpack is provided without any warranty ...                                                                                                                                 
                                                                                                                                                                                
                                                                                                                                                                                
                                                                                                                                                                                
PLease only ship the CRAN template for the BSD_3_clause license as file                                                                                                         
LICENSE.                                                                                                                                                                        
                                                                                                                                                                                
                                                                                                                                                                                
                                                                                                                                                                                
  The Description field should not start with the package name,                                                                                                                 
    'This package' or similar.                                                                                                                                                  
                                                                                                                                                                                
Flavor: r-devel-windows-ix86+x86_64                                                                                                                                             
Check: installed package size, Result: NOTE                                                                                                                                     
    installed size is 54.5Mb                                                                                                                                                    
    sub-directories of 1Mb or more:                                                                                                                                             
      libs  54.1Mb                                                                                                                                                              
                                                                                                                                                                                
This seems rather excessive?                                                                                                                                                    
                                                                                                                                                                                
                                                                                                                                                                                
Flavor: r-devel-linux-x86_64-debian-gcc                                                                                                                                         
Check: whether package can be installed, Result: FAIL                                                                                                                           
                                                                                                                                                                                
                                                                                                                                                                                
Any idea why this fails under Debian?                                                                                                                                           
                                                                                                                                                                                
                                                                                                                                                                                
Please fix and resubmit.                                                                                                                                                        
                                                                                                                                                                                
                                                                                                                                                                                
Best,                                                                                                                                                                           
Uwe Ligges

On 07.09.2020 22:59, CRAN submission wrote:                                                                                                                                     
> [This was generated from CRAN.R-project.org/submit.html]                                                                                                                      
>                                                                                                                                                                               
> The following package was uploaded to CRAN:                                                                                                                                   
> ===========================================                                                                                                                                   
>                                                                                                                                                                               
> Package Information:                                                                                                                                                          
> Package: mlpack                                                                                                                                                               
> Version: 3.4.1                                                                                                                                                                
> Title: 'Rcpp' Integration for the 'mlpack' Library                                                                                                                            
> Author(s): mlpack Team                                                                                                                                                        
> Maintainer: Ryan Curtin <ryan@ratml.org>                                                                                                                                      
> Depends: R (>= 4.0.0)                                                                                                                                                         
> Suggests: testthat (>= 2.1.0)                                                                                                                                                 
> Description: 'mlpack' is a fast, flexible machine learning library,                                                                                                           
>    written in C++, that aims to provide fast, extensible                                                                                                                      
>    implementations of cutting-edge machine learning algorithms.                                                                                                               
> License: BSD_3_clause + file LICENSE                                                                                                                                          
> Imports: Rcpp (>= 0.12.12)                                                                                                                                                    
> LinkingTo: Rcpp, RcppArmadillo (>= 0.8.400.0), BH (>= 1.58),                                                                                                                  
>    RcppEnsmallen (>= 0.2.10.0)                                                                                                                                                
>                                                                                                                                                                               
>                                                                                                                                                                               
> The maintainer confirms that he or she                                                                                                                                        
> has read and agrees to the CRAN policies.                                                                                                                                     
>                                                                                                                                                                               
> Submitter's comment: Hello there,                                                                                                                                             
>                                                                                                                                                                               
> This is a better (automatically                                                                                                                                               
>    generated) version of the RcppMLPACK package, now                                                                                                                          
>    called just "mlpack".  These are bindings to mlpack's                                                                                                                      
>    fast C++ implementations of machine learning                                                                                                                               
>    techniques, usable easily from R.                                                                                                                                          
>                                                                                                                                                                               
> This was done as                                                                                                                                                              
>    a Google Summer of Code project by Yashwant Singh,                                                                                                                         
>    mentored by myself, Dirk Eddelbuettel, and James                                                                                                                           
>    Balamuta.                                                                                                                                                                  
>                                                                                                                                                                               
> When I ran `R CMD check --as-cran`, I                                                                                                                                         
>    found that there were two NOTEs; one is about the                                                                                                                          
>    license in the DESCRIPTION and I am not sure that                                                                                                                          
>    anything needs to be done there---the license is                                                                                                                           
>    accurate (BSD 3-clause).  The second NOTE concerns                                                                                                                         
>    the large size of the package, which unfortunately is                                                                                                                      
>    unavoidable as much must be compiled upon                                                                                                                                  
>    installation.                                                                                                                                                              
>                                                                                                                                                                               
> I have read through the CRAN                                                                                                                                                  
>    policies and have done my best to follow them, but                                                                                                                         
>    please let me know if I missed anything---I am new to                                                                                                                      
>    the R community and have not submitted before (but                                                                                                                         
>    got nominated as the maintainer and contact                                                                                                                                
>    :-D).                                                                                                                                                                      
>                                                                                                                                                                               
> Thanks!                                                                                                                                                                       
>                                                                                                                                                                               
> =================================================                                                                                                                             
>                                                                                                                                                                               
> Original content of DESCRIPTION file:                                                                                                                                         
>                                                                                                                                                                               
> Package: mlpack                                                                                                                                                               
> Title: 'Rcpp' Integration for the 'mlpack' Library                                                                                                                            
> Version: 3.4.1                                                                                                                                                                
> Date: 2020-09-07                                                                                                                                                              
> Author: mlpack Team                                                                                                                                                           
> Maintainer: Ryan Curtin <ryan@ratml.org>                                                                                                                                      
> Description: 'mlpack' is a fast, flexible machine learning library, written                                                                                                   
>               in C++, that aims to provide fast, extensible implementations of                                                                                                
>               cutting-edge machine learning algorithms.                                                                                                                       
> SystemRequirements: A C++11 compiler. Versions 4.8.*, 4.9.* or later of                                                                                                       
>          GCC will be fine.                                                                                                                                                    
> License: BSD_3_clause + file LICENSE                                                                                                                                          
> Depends: R (>= 4.0.0)                                                                                                                                                         
> Imports: Rcpp (>= 0.12.12)                                                                                                                                                    
> LinkingTo: Rcpp, RcppArmadillo (>= 0.8.400.0), BH (>= 1.58),                                                                                                                  
>          RcppEnsmallen (>= 0.2.10.0)                                                                                                                                          
> Suggests: testthat (>= 2.1.0)                                                                                                                                                 
> URL: https://www.mlpack.org/doc/mlpack-3.4.1/r_documentation.html,                                                                                                            
>          https://github.com/mlpack/mlpack                                                                                                                                     
> BugReports: https://github.com/mlpack/mlpack/issues                                                                                                                           
> LazyData: true                                                                                                                                                                
> RoxygenNote: 7.1.1                                                                                                                                                            
> Encoding: UTF-8                                                                                                                                                               
> NeedsCompilation: yes                                                                                                                                                         
> Packaged: 2020-09-07 17:12:24 UTC; runner                                                                                                                                     
>
```

I responded:

```
Date: Wed, 9 Sep 2020 09:17:38 -0400                                                                                                                                            
From: Ryan Curtin <ryan@ratml.org>                                                                                                                                              
To: Uwe Ligges <ligges@statistik.tu-dortmund.de>                                                                                                                                
Cc: CRAN <cran-submissions@r-project.org>                                                                                                                                       
Subject: Re: CRAN submission mlpack 3.4.1                                                                                                                                       
User-Agent: Mutt/1.10.1 (2018-07-13)                                                                                                                                            
                                                                                                                                                                                
On Wed, Sep 09, 2020 at 09:04:57AM +0200, Uwe Ligges wrote:                                                                                                                     
> Thanks, we see:                                                                                                                                                               
>                                                                                                                                                                               
>   License components with restrictions and base license permitting such:                                                                                                      
>     BSD_3_clause + file LICENSE                                                                                                                                               
>   File 'LICENSE':                                                                                                                                                             
>     mlpack is provided without any warranty ...                                                                                                                               
>                                                                                                                                                                               
>                                                                                                                                                                               
>                                                                                                                                                                               
> PLease only ship the CRAN template for the BSD_3_clause license as file                                                                                                       
> LICENSE.                                                                                                                                                                      
                                                                                                                                                                                
Hi Uwe,                                                                                                                                                                         
                                                                                                                                                                                
Thanks for the quick response.  Sure, I can update to include only the                                                                                                          
BSD 3-clause license, although as the included license says, mlpack does                                                                                                        
include a couple components that are licensed under slightly different                                                                                                          
terms (MPL, BSL).                                                                                                                                                               
                                                                                                                                                                                
>   The Description field should not start with the package name,                                                                                                               
>     'This package' or similar.                                                                                                                                                
                                                                                                                                                                                
Sorry, I can fix this too.                                                                                                                                                      
                                                                                                                                                                                
> Flavor: r-devel-windows-ix86+x86_64                                                                                                                                           
> Check: installed package size, Result: NOTE                                                                                                                                   
>     installed size is 54.5Mb                                                                                                                                                  
>     sub-directories of 1Mb or more:                                                                                                                                           
>       libs  54.1Mb                                                                                                                                                            
>                                                                                                                                                                               
> This seems rather excessive?                                                                                                                                                  
                                                                                                                                                                                
This actually isn't excessive.  mlpack is a large C++ library containing                                                                                                        
over 40 machine learning techniques, and our R binding system requires a                                                                                                        
shared object for each individual binding.  So the size of the sources                                                                                                          
is ~1MB but when it's compiled for installation on a machine, it gets                                                                                                           
much larger.  A decent comparison here is the older RcppMLPACK package,                                                                                                         
which also will be large, but probably not as large as it does not                                                                                                              
include all of the bindings available in this package.                                                                                                                          
                                                                                                                                                                                
Hopefully the size is not too much of an issue---there's not really a                                                                                                           
way we can get around it, unfortunately.                                                                                                                                        
                                                                                                                                                                                
> Flavor: r-devel-linux-x86_64-debian-gcc                                                                                                                                       
> Check: whether package can be installed, Result: FAIL                                                                                                                         
>                                                                                                                                                                               
>                                                                                                                                                                               
> Any idea why this fails under Debian?                                                                                                                                         
                                                                                                                                                                                
It works locally---do you have any build log or anything?  I would not                                                                                                          
be surprised if it times out or something.                                                                                                                                      
                                                                                                                                                                                
> Please fix and resubmit.                                                                                                                                                      
                                                                                                                                                                                
Let me know your thoughts about the Debian failure and large size of the                                                                                                        
installed package, and then I'll apply the rest of the changes you                                                                                                              
suggested and resubmit.                                                                                                                                                         
                                                                                                                                                                                
Thanks!                                                                                                                                                                         
                                                                                                                                                                                
Ryan                                                                                                                                                                            
                                                                                                                                                                                
--                                                                                                                                                                              
Ryan Curtin    | "I am a meat popsicle."                                                                                                                                        
ryan@ratml.org |   - Korben Dallas
```

**Comment by eddelbuettel** [comment]
Ok, we can disentangle them.  1) is easiest. In the DESCRIPTION file use `License: BSD_3_clause + file LICENSE` and the include a file `LICENSE` contained (your adapted content, I just sketched this)

```
YEAR: 2007-2020
COPYRIGHT HOLDER: Ryan Curtin
```

(and I think you make that a longer enumerated list of authors as in the papers and also add 'and mlpack contrinutors')

The 'size seems rather excessive' is explainable; maybe include a rough line count of the C++ code.  Maybe also compare (a standard Linux build) with and without stripping -- not your fault "they" insist on debug symbols (which IIRC they still do, I strip locally)

We can look into whatever build fail (would surprise me) via RHub and its build.r-hub.io machine (you probably want to register there; just an email handshake to get a token id'ing you registered).

More later?  Maybe run replies off by us before shooting them off.  Ten fifteen years ago they replied immediately.  Now they are so snowed in that it can take days so can pay to make the emails "good" (I am bad that myself).

**Comment by rcurtin** [comment]
I'd really prefer not to register another account.  Remember I don't even speak R (and honestly it is not high on my list of languages to pick up---there's just not enough time)... is there any chance you are willing to look into whatever the build failure is, or get access to a log somehow?  My best guess is that it's running out of resources for the build, or timing out.  Perhaps Uwe will provide some more input.

**Comment by eddelbuettel** [comment]
Uwe deals with over 16k packages and (my guess) over 1k authors and I do not know how many daily submissions.  

And I misspoke. It is not "an account".  It is just verifying that your ratml.org email points to you after which you win-builder access which is scripted and covers 12 systems.  I think you would benefit from having it esp as reports only go to the listed "owner" of the package (by default Maintainer: ... unless overriden on upload).

**Comment by Yashwants19** [comment]
I am really sorry, I totally missed this [#comment](https://github.com/mlpack/mlpack/issues/2599#issuecomment-689603318) but I tried to check the package on `Debian Linux, R-devel, GCC` by changing the Maintainer's name (I hope this will not create any problem with CRAN re-submission.). These are the results:
[text](https://builder.r-hub.io/status/original/mlpack_3.4.1.tar.gz-4ba9d124eea04cbf844d557b1290b389), [HTML](https://builder.r-hub.io/status/mlpack_3.4.1.tar.gz-4ba9d124eea04cbf844d557b1290b389) (Here the package is installing properly, but due "Build time out" check is incomplete/Aborted.)

Do `CRAN` also have some time limit for building/checking the package?

**Comment by eddelbuettel** [comment]
Yes, examples and tests should not be too long.  I thought it was one minute each, but a few more are tolerated.  We should not run "all".  There are ways to toggle that.

Unlikely to be a rejection reason, may be a comment though.

**Comment by rcurtin** [comment]
@Yashwants19 do you want to take over the maintainer position?  You are way more knowledgeable than me, and my time to look into this is pretty limited.

**Comment by Yashwants19** [comment]
> Do CRAN also have some time limit for building/checking the package?

I am really sorry, I didn't mention this properly, my intention was to ask -- Do `CRAN` also have some time limit for installing(or No output after some time --> Abort) for the package? 

> Yes, examples and tests should not be too long. I thought it was one minute each, but a few more are tolerated. We should not run "all". There are ways to toggle that.

I guess I embed all the examples in `\dontrun{}` and we have only one test that also take 1.2 sec. I am really sorry if I misunderstanding something here.

**Comment by Yashwants19** [comment]
> @Yashwants19 do you want to take over the maintainer position? You are way more knowledgeable than me, and my time to look into this is pretty limited.

Hi @rcurtin I didn't take over the position of maintainer for the package, I just renamed it for running the `R CMD check` on rhub as I have done handshake before.

**Comment by Yashwants19** [comment]
> No time limit on installation or builds AFAIK. Some packages take 15 to 20 mins scream_cat

Then I guess we may have to wait for the reply from Uwe only, as the package is installing perfectly on r-hub and if I am not wrong the comment from Uwe was related to some installation only.

**Comment by rcurtin** [comment]
I don't understand what is required for the `LICENSE`.  I don't have copyright for the whole library, and not everything that is distributed is under the BSD 3-clause license.  So, as far as I can tell, what's currently written is accurate in `DESCRIPTION`:

```
License: BSD_3_clause + file LICENSE
```

And in the `LICENSE` file it explains the terms that are above and beyond the BSD 3-clause license.  I'm not sure what Uwe is looking for there?  Does he mean I should just remove the BSD 3-clause license from `LICENSE` and leave the rest there?

The other problem is the build timeout.  I said in the submission notes that the package is very large and should be expected to take a very long time.  How should we proceed with that part?  ...did my submission notes get tl;dr'ed?

I think those are the only two issues to address before resubmission.  At this point I am not expecting a response from Uwe.

Also, I really do think I am not the right person to be the maintainer of this package.  After all, this is basically a situation where everyone else is telling me what to do and I'm just executing commands.  If deeper issues arise during the packaging submission process, I'm not very familiar with the details and am likely to end up lost. :smile:  So... if anyone else is willing... I'd really appreciate it... :)

**Comment by coatless** [comment]
@rcurtin how about either Dirk or myself takes it over for the package submission phase and, then, once on CRAN we pass the maintainer role back to you?

**Comment by eddelbuettel** [comment]
> I don't understand what is required for the LICENSE

I am currently in three week dogfight with them over that for a _much_ smaller and clearer package.  It's gotten pretty beaureaucratic.  As I run service monitoring new and update package publishing [1], I noticed the copyright statements in DESCRIPTION seem to be getting more verbose.  Total pain in the rear.

Don't get me wrong. I am (obviously) very much in favour of giving credit, citing properly, respecting licenses. But there is time-honoured way to do this, and a completely painful of obsessive way.  Seems we turned direction.  Not happy.  Obviously I still oblige (what other choice do I have?) but with that ... expect some more back and forth.

[1] At https://dirk.eddelbuettel.com/cranberries/ and then also on Twitter as @CRANberriesFeed

**Comment by rcurtin** [comment]
@coatless I would appreciate that *so much*.  I'm also I'm not sure I'm the right person for long-term maintenance, but I'm willing to take it over because I'm hopeful we can get maintenance down to something very simple like these guidelines we have for Go and Julia:

https://github.com/mlpack/mlpack.jl/blob/master/rel/deployment.md
https://github.com/mlpack/mlpack-go/blob/master/rel/deployment.md

@eddelbuettel ack, that seems frustrating. :(  We do have some options.  The only reason the license is not "just" BSD 3-clause is these three bits, which I can provide some clarity on each of:

```
mlpack contains some reproductions of the source code of Armadillo, which is
licensed under the Mozilla Public License v2.0 (MPL2).  This code is found in
src/mlpack/core/arma_extend/ and more details on the licensing are available
there.
```

I believe as part of #2458 and #2602, this will no longer be true.  However we're not quite there yet.

```
mlpack also contains some reproductions of the source code of Boost, which is
licensed under the Boost Software License, version 1.0.  This code is found in
src/mlpack/core/boost_backport/ and more details on the licensing are available
there.
```

As long as we're using boost::serialization this is true, but we can avoid the other stuff if R has an acceptably new version of boost.  This will not be true forever, though.

```
mlpack may contain some usage of the source code of stb, which is licensed~
under the MIT License and the Public Domain (www.unlicense.org). This code
is used in src/mlpack/core/data/load_image.hpp.
```

I don't think that we can get around this---some of the bindings use STB, and unless it's available via R (probably not?) this part will have to stay.

So I'm not sure exactly what to do on those parts, or if there's an easy way around the bureaucracy.  To be technically legally correct though, we do have to reproduce those clauses with our distribution.

**Comment by eddelbuettel** [comment]
If James wants to shepherd it that would be great. I am not quite in the mood for it having also fought a recent fight with the 'for work' package...

**Comment by coatless** [comment]
More than happy to. @rcurtin can you forward me the last e-mail you sent to CRAN? I'd like Uwe to know that I'm taking it over.

**Comment by rcurtin** [comment]
Thank you so much @coatless.  Just forwarded the email. :+1:

**Comment by rcurtin** [comment]
Ahh, I just got a DSN from my response to Uwe.  Turns out his mailserver rejected my email, so, this explains the lack of response. :smile:  (reverse DNS was not set up for my IP since my hosting company changed it... I think I will be getting that fixed basically immediately!)

**Comment by coatless** [comment]
@rcurtin I haven't heard back since I sent an e-mail to him/CRAN on Friday. I'll probably prepare another release for CRAN on Wednesday.

**Comment by eddelbuettel** [comment]
@coatless Let me know if you need a hand with anything.  CRAN is extra slow these days so no rush.  (But [RcppSpdlog](https://github.com/eddelbuettel/rcppspdlog) is finally on, yay!)

**Comment by eddelbuettel** [comment]
@coatless Any news from your side?  Do you want me to take a swing at it this weekend?

**Comment by eddelbuettel** [comment]
How are we doing here?

**Comment by rcurtin** [comment]
I haven't heard anything back.  @coatless let us know how we can be helpful. :+1:

**Comment by eddelbuettel** [comment]
Hm, how about if we attempted a new upload with @rcurtin as maintainer as help as we can from the sides? I am sure @coatless will happily pitch in, time permitting, but doing so would allow us to multithread a little better than the current wait deadlock.

**Comment by rcurtin** [comment]
@eddelbuettel sure, I just submitted it to CRAN.  @coatless sorry I didn't have time to do this earlier and asked for some help then---but things have freed up for me again and I can take on the task.  I am worried about this being a maintenance burden (especially since I don't speak R :smile:) but if we can get a system set up where I just send the tarball off to CRAN after each release, I can do that (and I can probably automate it too :)).

I had to change a few things in the package, but we can fix these in our codebase once the package is accepted:

 * Replace `LICENSE` just with the template.
 * The version in the tarball on the 3.4.2 tag was labeled `3.4.3`, so I had to fix the version (easy fix).
 * I updated the description so it doesn't start with the name of the package.
